pypesto/optimize/optimizer.py

```diff
@@ -309,6 +309,14 @@ def minimize(
                     "For least squares optimization, the objective "
                     "must be able to compute residuals."
                 )
+            # raise a warning, if priors are used in least squares
+            if isinstance(objective, (NegLogParameterPriors, NegLogPriors)):
+                logger.warning(
+                    "Priors are used together with SciPy least squares. "
```

Comment:
I would rather add such a warning (probably independently of prior or not) to the ScipyOptimizer docstring.

And for the derivative of the Laplace prior, filter the following warning and replace it by a more informative warning(like: Laplace not differentiable at the mean):
`pypesto/objective/priors.py:493: RuntimeWarning: invalid value encountered in scalar divide
    return 1 / 2 * (x - mean) / np.sqrt(scale * abs(x - mean) ** 3)`

Reply:
ahh good idea. 👍🏼

Reply:
wouldn't it rather be a `ValueError`?

Reply:
> wouldn't it rather be a `ValueError`?

One could argue for either. Could be simply a `nan` with or without a warning, because that's simply what you usually get if you divide by zero; could be a ValueError if the documentation explicitly excludes x==mean as valid inputs (it currently doesn't); could be `ZeroDivisionError`, because that's what Python raises for a plain `1/0`. 

Elsewhere we use NaN (for fval) in case of simulation errors. I would argue for sticking to that. An optimizer might recognise NaN as "cannot evaluate at this point" and try to recover,  whereas a ValueError would probably result in immediate termination. Then again, this will depend on the optimizer implementation.